File: bindings/lua/Polycode/DebugBackTraceEntry.lua
class "DebugBackTraceEntry"


function DebugBackTraceEntry:__getvar(name)
	if name == "fileName" then
		return Polycode.DebugBackTraceEntry_get_fileName(self.__ptr)
	elseif name == "lineNumber" then
		return Polycode.DebugBackTraceEntry_get_lineNumber(self.__ptr)
	end
end

function DebugBackTraceEntry:__setvar(name,value)
	if name == "fileName" then
		Polycode.DebugBackTraceEntry_set_fileName(self.__ptr, value)
		return true
	elseif name == "lineNumber" then
		Polycode.DebugBackTraceEntry_set_lineNumber(self.__ptr, value)
		return true
	end
	return false
end
function DebugBackTraceEntry:__delete()
	if self then Polycode.delete_DebugBackTraceEntry(self.__ptr) end
end

File: bindings/javascript/Polycode/DebugBackTraceEntry.js
function DebugBackTraceEntry() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.DebugBackTraceEntry()
	}
	Object.defineProperties(this, {
		'fileName': { enumerable: true, configurable: true, get: DebugBackTraceEntry.prototype.__get_fileName, set: DebugBackTraceEntry.prototype.__set_fileName},
		'lineNumber': { enumerable: true, configurable: true, get: DebugBackTraceEntry.prototype.__get_lineNumber, set: DebugBackTraceEntry.prototype.__set_lineNumber}
	})
}

DebugBackTraceEntry.prototype.__get_fileName = function() {
	return Polycode.DebugBackTraceEntry__get_fileName(this.__ptr)
}

DebugBackTraceEntry.prototype.__set_fileName = function(val) {
	Polycode.DebugBackTraceEntry__set_fileName(this.__ptr, val)
}

DebugBackTraceEntry.prototype.__get_lineNumber = function() {
	return Polycode.DebugBackTraceEntry__get_lineNumber(this.__ptr)
}

DebugBackTraceEntry.prototype.__set_lineNumber = function(val) {
	Polycode.DebugBackTraceEntry__set_lineNumber(this.__ptr, val)
}

Duktape.fin(DebugBackTraceEntry.prototype, function (x) {
	if (x === DebugBackTraceEntry.prototype) {
		return;
	}
	Polycode.DebugBackTraceEntry__delete(x.__ptr)
})
